TypeCobol/Compiler/CodeElements/Expressions/SymbolName.cs

```diff
@@ -117,13 +117,16 @@ public override bool AcceptASTVisitor(IASTVisitor astVisitor)
     /// <summary>
     /// Reference to a previously defined symbol in the Cobol syntax
     /// </summary>
-    public class SymbolReference : SymbolInformation
+    public class SymbolReference : SymbolInformation, IEquatable<SymbolReference>
     {
+        private readonly int _hashCode;
+
         public SymbolReference(SyntaxValue<string> nameLiteral, SymbolType type) :
             base(nameLiteral, SymbolRole.SymbolReference, type)
         {
             IsAmbiguous = false;
             IsQualifiedReference = false;
+            _hashCode = nameLiteral.GetHashCode() + type.GetHashCode();
```

Comment:
Why use a cache value for _hashCode in this class and not the others ?

Reply:
The reason was _hashCode is a readonly field; Type is a non-readonly property.
The call to GetHashCode needs a readonly field/property.
Modification has been made to use "Type" and "Name" changed in readonly properties.
https://github.com/TypeCobolTeam/TypeCobol/pull/1555/commits/77c907c8055e95bf877861ca4745d1033758792c